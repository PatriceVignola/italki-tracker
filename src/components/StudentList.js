/**
 * @prettier
 * @flow
 */

import React from 'react';
import {Grid, Typography} from '@material-ui/core';

import StudentCard from './StudentCard';
import AddStudentButton from './AddStudentButton';
import type {Student} from '../actions/types';

type Props = {
  students: Student[],
};

function StudentList(props: Props) {
  return (
    <div style={styles.container}>
      <Typography variant="title" style={styles.title}>
        Students
      </Typography>
      <Grid container spacing={16} style={styles.grid}>
        {props.students.map(student => (
          <StudentCard key={student.id} student={student} />
        ))}
      </Grid>
      <AddStudentButton style={styles.addButton} />
    </div>
  );
}

const styles = {
  container: {
    flex: 1,
  },
  title: {
    marginTop: 25,
    marginBottom: 25,
  },
  grid: {
    margin: 5,
  },
  addButton: {
    position: 'absolute',
    bottom: 24,
    right: 24,
  },
};

export default StudentList;
