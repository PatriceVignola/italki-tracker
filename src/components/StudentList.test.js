/**
 * @prettier
 * @flow
 */

import React from 'react';
import {shallow} from 'enzyme';

import StudentList from './StudentList';

describe('StudentList', () => {
  it('renders correctly', () => {
    const mockStudents = [
      {
        id: 1,
        firstName: 'FirstName',
        lastName: 'LastName',
        photoUrl:
          'http://education.mnhs.org/immigration/sites/education.mnhs.org.immigration/files/imagecache/Full_800x800/MaleSilhouette_5.png',
      },
    ];

    const wrapper = shallow(<StudentList students={mockStudents} />);
    expect(wrapper).toMatchSnapshot();
  });
});
