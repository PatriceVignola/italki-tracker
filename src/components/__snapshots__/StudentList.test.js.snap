// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentList renders correctly 1`] = `
<div
  style={
    Object {
      "flex": 1,
    }
  }
>
  <WithStyles(Typography)
    style={
      Object {
        "marginBottom": 25,
        "marginTop": 25,
      }
    }
    variant="title"
  >
    Students
  </WithStyles(Typography)>
  <GridWrapper
    container={true}
    spacing={16}
    style={
      Object {
        "margin": 5,
      }
    }
  >
    <StudentCard
      key="1"
      student={
        Object {
          "firstName": "FirstName",
          "id": 1,
          "lastName": "LastName",
          "photoUrl": "http://education.mnhs.org/immigration/sites/education.mnhs.org.immigration/files/imagecache/Full_800x800/MaleSilhouette_5.png",
        }
      }
    />
  </GridWrapper>
  <AddStudentButton
    style={
      Object {
        "bottom": 24,
        "position": "absolute",
        "right": 24,
      }
    }
  />
</div>
`;
